TiFBackendUtils/SQLExecutable/index.ts

```diff
@@ -1,60 +1,67 @@
-import dotenv from "dotenv"
-import mysql from "mysql2/promise.js"
-import { z } from "zod"
-import { SQLExecutable } from "./utils.js"
-export { SQLExecutable }
+import dotenv from 'dotenv';
+import mysql from 'mysql2/promise.js';
+import { z } from 'zod';
+import { SQLExecutable } from './utils.js';
 
-dotenv.config()
+export { SQLExecutable };
 
-const EnvVarsSchema = z
-  .object({
-    DATABASE_HOST: z.string(),
-    DATABASE_PASSWORD: z.string(),
-    DATABASE_USERNAME: z.string()
-  })
-  .passthrough()
+dotenv.config();
 
-const envVars = EnvVarsSchema.parse(process.env)
+const EnvVarsSchema = z.object({
+  DATABASE_HOST: z.string(),
+  DATABASE_NAME: z.string(),
+  DATABASE_PASSWORD: z.string(),
+  DATABASE_USERNAME: z.string(),
+}).passthrough();
 
-const dbName = 'tif'
+const envVars = EnvVarsSchema.parse(process.env);
+
+let connection: mysql.Connection;
```

Comment:
I think this would better off be a `Promise` variable inside the `SQLExecutable` class. You could have something like:

```ts
class SQLExecutable {
  private readonly connectionPromise: Promise<mysql.Connection>

  constructor(connect: () => Promise<mysql.Connection>) {
    this.connectionPromise = connect()
  }

  async query(sql: string) {
    const connection = await this.connectionPromise
    // Use connection to perform query...
  }

  // ...
}
```

Reply:
Tried this, it causes an open handle in tests.